test_requests.py

```diff
@@ -188,6 +188,28 @@ def test_HTTP_302_ALLOW_REDIRECT_GET(self, httpbin):
         assert r.history[0].status_code == 302
         assert r.history[0].is_redirect
 
+    def test_HTTP_302_TOO_MANY_REDIRECTS(self, httpbin):
+        try:
+            requests.get(httpbin('redirect', '50'))
+        except TooManyRedirects as e:
+            assert '/relative-redirect/20' in e.request.url
+            assert e.request.url == e.response.url
```

Comment:
On January 20, 2016 1:14:50 PM CST, Ryan Munro <notifications@github.com> wrote:
>> @@ -188,6 +188,28 @@ def test_HTTP_302_ALLOW_REDIRECT_GET(self,
>httpbin):
>>          assert r.history[0].status_code == 302
>>          assert r.history[0].is_redirect
>>  
>> +    def test_HTTP_302_TOO_MANY_REDIRECTS(self, httpbin):
>> +        try:
>> +            requests.get(httpbin('redirect', '50'))
>> +        except TooManyRedirects as e:
>> +            assert '/relative-redirect/20' in e.request.url
>> +            assert e.request.url == e.response.url
>
>I think it could help catch off by one logic where the previous
>redirect request is returned instead of the one used to get the last
>request.  I'll swap it for a more direct comparison instead.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/kennethreitz/requests/pull/2972/files#r50303029

What about e.request is e.response.request? I still don't think this is necessary but that's a more clear check. Could also use a comment.
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.